chain-vote/src/lib.rs

```diff
@@ -124,6 +124,70 @@ impl Tally {
         }
         (TallyState { r2s }, TallyDecryptShare { r1s: dshares })
     }
+
+    pub fn to_bytes(&self) -> Vec<u8> {
+        use std::io::Write;
+        let mut bytes: Vec<u8> = Vec::with_capacity(130 * self.r.len());
+        for ri in &self.r {
+            bytes.write(ri.to_bytes().as_ref()).unwrap();
+        }
+        bytes
+    }
+
+    pub fn from_bytes(bytes: &[u8]) -> Option<Self> {
+        if bytes.len() % 130 != 0 {
+            return None;
+        }
+        let len = bytes.len() / 130;
+        let mut r: Vec<Ciphertext> = Vec::with_capacity(len);
+        for i in 0..len {
+            let ri = Ciphertext::from_bytes(&bytes[(i * 130)..((i + 1) * 130)])?;
+            r.push(ri);
+        }
+        Some(Self { r })
+    }
+}
+
+impl TallyDecryptShare {
+    pub fn to_bytes(&self) -> Vec<u8> {
+        group_elements_to_bytes(&self.r1s)
+    }
+
+    pub fn from_bytes(bytes: &[u8]) -> Option<Self> {
+        group_elements_from_bytes(bytes).map(|r1s| Self { r1s })
+    }
+}
+
+impl TallyState {
+    pub fn to_bytes(&self) -> Vec<u8> {
+        group_elements_to_bytes(&self.r2s)
+    }
+
+    pub fn from_bytes(bytes: &[u8]) -> Option<Self> {
+        group_elements_from_bytes(bytes).map(|r2s| Self { r2s })
+    }
+}
+
+fn group_elements_to_bytes(elements: &[gang::GroupElement]) -> Vec<u8> {
+    use std::io::Write;
+    let mut bytes: Vec<u8> = Vec::with_capacity(65 * elements.len());
```

Comment:
I don't like this magic number everywhere... Is it fixed in the algorithm and can never change later?

Reply:
Yeah, size of `GroupElement` is fixed size. I think we can have an associated constant for it.